### ISSUE TITLE
[MNG-5113] NullPointerException on javadoc site generation

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7046]
**[Christian](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=palaga)** opened **[MNG-5113](https://issues.apache.org/jira/browse/MNG-5113?redirect=false)** and commented

Dear codehaus developers,

Today I encountered a pretty serious error, while trying to generate javadocs for my code. This was the first time I added the reporting part to my pom. Site generation worked fine without this part. After I added this part, maven gave me a NullPointerException.
It was very easy for me to reproduce the bug. I created an empty directory and put the following pom there:

```xml
<?xml version="1.0"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>Test</groupId>
  <artifactId>test</artifactId>
  <version>1.0-SNAPSHOT</version>

  <reporting>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artefactId>maven-javadoc-plugin</artefactId>
        <version>2.8</version>
      </plugin>
    </plugins>
  </reporting>
</project>
```

Then I ran:

```
mvn site
```

I've attached the debugging output and this example pom to this report.

Best regards,

Chris

---

**Affects:** 3.0.3

**Attachments:**
- [debug.log](https://issues.apache.org/jira/secure/attachment/12713258/debug.log) (_2.83 kB_)
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12712686/pom.xml) (_598 bytes_)

**Comment by jira-importer** [comment]
**[Christian](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=palaga)** commented

Ok, all this was caused by a little typo. I typed artefact instead of artifact. I didn't noticed that, at the time of writing. After I submitted the bug, I added the plugin to the build plugin list, this generated an error, saying that I had a typo in my pom. So this was all by fault.
Still, I'll leave the issue open, since I the NullPointerException thrown, shouldn't have been thrown. This should've been the typo-error (Non-parsable POM), which I got after adding the plugin to the build section.

Chris.
